x-pack/libbeat/management/generate.go

```diff
@@ -94,7 +94,7 @@ func CreateInputsFromStreams(raw *proto.UnitExpectedConfig, inputType string, ag
 		}
 
 		// 3. stream processors
-		streamSource, err = injectStreamProcessors(raw, inputType, stream, streamSource)
+		streamSource, err = injectStreamProcessors(raw, inputType, stream, streamSource, defaultProcessors)
```

Comment:
There is this above:

```go
		// 1. global processors
		streamSource = injectGlobalProcesssors(raw, streamSource)
```

The metadata processors behave more like global processors don't they? Shouldn't they be added there? 

Both prepend and append are used here to control the order so just reading this code it is a bit tricky to follow what ends up where.

Could you post an example `beat-rendered-config` if you have one from diagnostics with this change?

Reply:
yes but if you check the final order of the processors it's
```
		// now the order of the processors on this input is as follows
		// 1. stream processors
		// 2. agentInfo processors
		// 3. global processors
		// 4. stream specific processors
```

so if we put it with global, they will be pushed to later stage by next 2 steps. this is why i went with stream
- it makes sure it's first
- it makes clear that it will be part of the stream

Reply:
Got it, thanks.